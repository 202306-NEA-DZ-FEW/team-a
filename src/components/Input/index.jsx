function Input({
  name,
  placeholder,
  label,
  type,
  handleChange,
  value,
  handleBlur,
  touched,
  error,
}) {
  return (
    <div>
      <div className='label'>
        <label htmlFor={name} className='label-text'>
          {label}
        </label>
        <label
          className={`label-text-alt ${touched && error ? "text-error" : ""}`}
        >
          {touched && error ? error : ""}
        </label>
      </div>
      <input
        className='input bg-white input-primary w-full'
        id={name}
        type={type}
        name={name}
        placeholder={placeholder}
        onChange={handleChange}
        value={value}
        onBlur={handleBlur}
      />
    </div>
  );
}

export default Input;
