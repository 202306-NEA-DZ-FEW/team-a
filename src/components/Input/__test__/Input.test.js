import renderer from "react-test-renderer";

import Input from "..";

it("renders correctly", () => {
  const tree = renderer.create(<Input />).toJSON();
  expect(tree).toMatchSnapshot();
});
