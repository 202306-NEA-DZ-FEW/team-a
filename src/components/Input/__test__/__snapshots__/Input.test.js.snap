// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="label"
  >
    <label
      className="label-text"
    />
    <label
      className="label-text-alt "
    />
  </div>
  <input
    className="input bg-white input-primary w-full"
  />
</div>
`;
